### PR TITLE
🤖 test: full-login validation matrix and loopback provisioner guard (LK-18 PR2)

### PR DESCRIPTION
**Pull request by PimvanderLoos**
## Why
- PR #100 shipped the full-login mechanism with smoke tests; the design's 13-test validation matrix (Appendix A) proves the pipeline behaves like a real client across denial, persistence, session, and plugin-interaction dimensions — and it already paid for itself by catching three real regressions in the merged driver.

## How
- Two IT classes (11 tests, both lanes): login-event order with tick stamps, ban/whitelist typed denials (deny→allow), rejoin identity + inventory persistence, real quit events, teleport-ack, compression, five simultaneous sessions, server-observed locale, chat parity, and LuckPerms Permissible-injection survival (LP provisioned from Modrinth, granted via console, asserted live).
- `LoopbackLoginGuard`: `prepare-server` now fails loudly (naming FULL_LOGIN) on `online-mode=true`, active BungeeCord, or Velocity forwarding — the configs that break loopback logins; case/spacing-tolerant parsing, runs after all overlays.
- `/lktestlocale` test-plugin observable for server-side locale assertions.
- Three production regression fixes the matrix exposed in the merged base: denial outcomes complete before the channel closes (kick reasons no longer masked by the close listener); full-login bots leave via the real disconnect flow (`kickPlayer`) with a fail-loud check for `PlayerKickEvent` vetoes — the old path left the TCP connection open and could crash Paper via double-removal; the driver now waits (bounded, fail-loud) for channel activation before sending hello, closing a stalled-login race.

## Tests
- `mvn --batch-mode -Perrorprone clean verify checkstyle:checkstyle pmd:check` → BUILD SUCCESS (full reactor; 49/49 ITs on both Paper and Spigot lanes).
- New unit tests: guard parsing edge cases (10), locale command executor (2), kick-veto and kick-routing paths in the agent (2).
- Pre-PR deep review applied (verdict gated on the kick-veto check — fixed): all three regression fixes' mechanical claims verified against both distros' bytecode.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved FULL_LOGIN connection reliability by waiting for the network channel to become ready before starting login.
  * Preserved specific server denial and connection failure reasons instead of replacing them with generic messages.
  * Ensured fully logged-in synthetic players disconnect through the normal server flow.
  * Improved cleanup when player removal is cancelled or interrupted.

* **Validation**
  * Server preparation now detects incompatible online-mode and proxy-forwarding settings before startup.

### CODE REVIEW

**Comment by PimvanderLoos** [comment]
@coderabbitai full review

**Comment by PimvanderLoos** [comment]
@codex review

File: lightkeeper-maven-plugin/src/main/java/nl/pim16aap2/lightkeeper/maven/provisioning/LoopbackLoginGuard.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    }
+
+    private static void validateServerProperties(Path serverPropertiesFile)
+        throws MojoExecutionException
+    {
+        for (final String line : readLinesIfPresent(serverPropertiesFile))
+        {
+            final String trimmed = line.trim();
+            if (trimmed.startsWith("#"))
+                continue;
```
[the remark this answers is not in the document]
Comment (PimvanderLoos) [reply lightkeeper-maven-plugin/src/main/java/nl/pim16aap2/lightkeeper/maven/provisioning/LoopbackLoginGuard.java:89]: Fixed in 83ab657: the guard now skips both '#' and '!' comment prefixes per the java.util.Properties spec, with the commented-settings test extended to cover the '!' spelling.

File: lightkeeper-integration-tests/src/test/java/nl/pim16aap2/lightkeeper/maven/test/LightkeeperFullLoginSessionIT.java [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+            // verify — the position is applied server-side and the session survives the ack round trip.
+            eventually(Duration.ofSeconds(10), () ->
+                assertThat(playerPosition(world, player.uniqueId())).hasValueSatisfying(position ->
+                    assertPositionCloseTo(position, target)));
+            player.andWaitTicks(20);
+            assertThat(eventsWithPlayerRef(quitCapture.getCapturedEvents(), player.uniqueId())).isEmpty();
+
+            player.remove();
+        }
+    }
```
[the remark this answers is not in the document]
Comment (PimvanderLoos) [reply lightkeeper-integration-tests/src/test/java/nl/pim16aap2/lightkeeper/maven/test/LightkeeperFullLoginSessionIT.java:78]: Declining: bot cleanup on failure is already guaranteed by the framework's per-method scope reaping (endMethodScope → removePlayer), which runs regardless of test outcome — the pre-PR deep review verified leaked bots are reaped between methods. Duplicating cleanup in every test would shadow the framework guarantee these ITs exist to exercise; if that reaping ever regressed, we'd rather these tests expose it loudly than mask it.
